CIP-XXXX/README.md

```diff
@@ -0,0 +1,181 @@
+
+```json
+    "Credential": {
+      "type": "object",
+      "discriminator": {
+        "propertyName": "tag"
+      },
+      "oneOf": [
+        {
+          "type": "object",
+          "properties": {
+            "tag": {
+              "enum": [
+                "pubkey_hash"
+              ]
+            },
+            "value": {
+              "$ref": "cardano-babbage.json#/definitions/Ed25519KeyHash"
+            }
+          },
+          "required": ["tag", "value"],
+          "additionalProperties": false
+        },
+        {
+          "type": "object",
+          "properties": {
+            "tag": {
+              "enum": [
+                "script_hash"
+              ]
+            },
+            "value": {
+              "$ref": "cardano-babbage.json#/definitions/ScriptHash"
+            }
+          },
+          "required": ["tag", "value"],
+          "additionalProperties": false
+        }
+      ]
+    },
+```
+
+Other properties of a tagged object MUST be specified in lower-case snake-case.
+
+#### Encoding of enum types
+
+Enums are a special kind of variant types that carry no payloads. These must be encoded as string `enum`s.
+
+Lowercase snake case identifiers should be used for the options, e.g.:
+
+```json
+    "Language": {
+      "title": "Language",
+      "type": "string",
+      "enum": [
+        "plutus_v1",
+        "plutus_v2"
+      ]
+    },
+```
+
+### Encoding of record types
+
+All record types MUST be encoded as objects with explicit list of `required` properties, and `additionalProperties` set to `false` (see "absence of extensibility" chapter for the motivation behind this suggestion).
+
+### Encoding of nominal type synonyms
+
+Some of the types have identical representations, differing only by nominal name. For example, Slot domain type is expressed as `uint` in CDDL.
+
+For these types, their nominal name should not have a separate definition in the json-schema, and the "representation type" should be used via a `$ref` instead. The domain type name SHOULD be included as `title` string at the point of usage.
+
+### Additional format types
+
+Some non-standard `format` types are used:
+
+- `hex`
+- `bech32`
+- `base58`
+
+TODO: describe the formats
+
+## Limitations
+
+### Byte length limits for strings
+
+In CDDL, the length of a `tstr` value gives the number of bytes, but in `json-schema` there is no way to specify restrictions on byte lengths. So, `maxLength` is not the correct way of specifying the limits, but it is still useful, because no string longer than 64 *characters* satisfies the 64-byte limit.
+
+## Rationale: how does this CIP achieve its goals?
+
+## Path to Active
+
+- [ ] Complete the specification
+- [ ] Provide an implementation of validating functions that uses this json-schema
+- [ ] Collect a list of cardano domain types implementations and negotiate transition to the specified formats with maintainers (if it makes sense and is possible)
+
+### Acceptance Criteria
+
```

Comment:
```suggestion

- [ ] Completed specifications for: Byron, Shelly, Allegra, Mary, Alonzo, Babbage and Conway.
- [ ] Have one future ledger era supported via subsequent CIP.
```

Reply:
Specifications for past eras are not needed: no one is going to deploy new software for these eras. There is nowhere to deploy, even.

I decided to start with babbage, because the cddl spec for Conway does not look finalized (yet).

https://github.com/cardano-foundation/CIPs/pull/742/files#diff-d1d319150c1890f55b035a33282c3b890d814360fff9778f6dd5208b7eb9ecb0R45

Reply:
Agreed!